pkg/run/install/install_fluent_bit.go

```diff
@@ -0,0 +1,247 @@
+package install
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"html/template"
+	"strings"
+	"time"
+
+	helmv2 "github.com/fluxcd/helm-controller/api/v2beta1"
+	sourcev1 "github.com/fluxcd/source-controller/api/v1beta2"
+	"github.com/weaveworks/weave-gitops/pkg/logger"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const configInputs = `
+[INPUT]
+    Name tail
+    Path /var/log/containers/*.log
+    multiline.parser docker, cri
+    Tag kube.*
+    Mem_Buf_Limit 5MB
+    Skip_Long_Lines Off
+`
+
+const configFilters = `
+[FILTER]
+    Name kubernetes
+    Match kube.*
+    Merge_Log On
+    Keep_Log Off
+    K8S-Logging.Parser On
+    K8S-Logging.Exclude On
+[FILTER]
+    Name    grep
+    Match   *
+    Exclude $kubernetes['namespace_name'] (gitops-run|kube-system)
+[FILTER]
+    Name    grep
+    Match   *
+    Exclude $kubernetes['pod_name'] ^fluent\-bit
+`
+
+func makeConfigOutputs(bucket string, port int32) (string, error) {
+	if port < 1 || port > 65535 {
+		return "", fmt.Errorf("port must be between 1 and 65535")
+	}
+
+	const configOutputs = `
+[OUTPUT]
+    Name s3
+    Match kube.*
+    bucket {{ .Bucket }}
+    endpoint http://run-dev-bucket.gitops-run.svc:{{ .Port }}
+    tls Off
+    tls.verify Off
+    use_put_object true
+    preserve_data_ordering true
+    static_file_path true
+    total_file_size 1M
+    upload_timeout 15s
+    s3_key_format /fluent-bit-logs/$TAG[4].%Y%m%d%H%M%S
+`
+	tmpl, err := template.New("configOutputs").Parse(strings.TrimSpace(configOutputs))
+	if err != nil {
+		return "", fmt.Errorf("error parsing template: %v", err)
+	}
+	var result strings.Builder
+	err = tmpl.Execute(&result, struct {
+		Bucket string
+		Port   int32
+	}{Bucket: bucket, Port: port})
+	if err != nil {
+		return "", fmt.Errorf("error executing template: %v", err)
+	}
+	return result.String(), nil
+}
+
+func mustBeString(s string, e error) string {
+	if e != nil {
+		panic(e)
```

Comment:
⚠️  we really should not panic if there is any way to avoid it. It makes the code nearly untestable.